### PR TITLE
modules/ignition,platforms/*: Optionally allow users to set NTP servers for the cluster

### PR DESCRIPTION
**Pull request by kalmog**
NTP servers on Container Linux default to CoreOS time servers,
which may not be accessible on private cluster deployments.
Allow users to set a list of primary NTP servers with the CoreOS
servers as the default fallback.

This is rebased onto track-1 branch from @lander2k2 PR: #2652

### CODE REVIEW

**Comment by kalmog** [comment]
ok to test

**Comment by kalmog** [comment]
test this please

**Comment by kalmog** [comment]
ok to test

**Comment by kalmog** [comment]
ok to test

File: modules/ignition/outputs.tf [hunk 1]
```diff
@@ -30,6 +30,14 @@ output "k8s_node_bootstrap_service_rendered" {
   value = "${data.template_file.k8s_node_bootstrap.rendered}"
 }
 
+output "ntp_dropin_id" {
+  value = "${data.ignition_file.ntp_dropin.id}"
+}
+
+output "ntp_dropin_rendered" {
+  value = "${data.template_file.ntp_dropin.rendered}"
```
[the remark this answers is not in the document]
Comment (kalmog) [reply modules/ignition/outputs.tf]: After removing the `count` var from the template file, this change is not needed.

File: modules/ignition/outputs.tf [hunk 2]
```diff
@@ -30,6 +30,14 @@ output "k8s_node_bootstrap_service_rendered" {
   value = "${data.template_file.k8s_node_bootstrap.rendered}"
 }
 
+output "ntp_dropin_id" {
+  value = "${data.ignition_file.ntp_dropin.id}"
```
[the remark this answers is not in the document]
Comment (kalmog) [reply modules/ignition/outputs.tf]: After removing the count var from the template file, this change is not needed.

**Comment by cpanato** [comment]
retest this please

**Comment by cpanato** [comment]
following this PR to react in test failures

**Comment by cpanato** [comment]
retest this please, AZURE/GOVCLOUD

**Comment by cpanato** [comment]
retest this please. aws/metal.  azure and gov are green

**Comment by carlosaya** [comment]
has this been tested with vmware at all?  I cannot terraform init.  Get the following:

`[root@ta030621-dev ite13]# ../../tectonic-installer/linux/terraform init ../../platforms/vmware
Downloading modules...
Error getting plugins: module root:
    5 error(s) occurred:

* module 'masters': unknown variable referenced: 'tectonic_ntp_servers'. define it with 'variable' blocks
* module 'ignition_masters': unknown variable referenced: 'tectonic_ntp_servers'. define it with 'variable' blocks
* module 'ignition_workers': unknown variable referenced: 'tectonic_ntp_servers'. define it with 'variable' blocks
* module 'workers': unknown variable referenced: 'tectonic_ntp_servers'. define it with 'variable' blocks
* module 'etcd': unknown variable referenced: 'tectonic_ntp_servers'. define it with 'variable' blocks
    module masters: ign_ntp_dropin_id is not a valid parameter
    module workers: ign_ntp_dropin_id is not a valid parameter
  module workers.root: 1 error(s) occurred:

* resource 'data.ignition_config.node' config: unknown variable referenced: 'ign_ntp_dropin_id'. define it with 'variable' blocksmodule masters.root: 1 error(s) occurred:

* resource 'data.ignition_config.node' config: unknown variable referenced: 'ign_ntp_dropin_id'. define it with 'variable' blocks
`

**Comment by coresolve** [comment]
@kalmog can you address the vmware question above?

**Comment by kalmog** [comment]
It was indeed tested. Are you sure you have a clean branch?
```karena@Karens-MacBook-Pro:~/go/src/github.com/coreos/tectonic-installer/build/kalmog (track-1) [ 10139283 ] $ terraform init ../../platforms/vmware
Initializing modules...
- module.etcd
- module.ignition_masters
- module.masters
- module.ignition_workers
- module.workers
- module.kube_certs
- module.etcd_certs
- module.ingress_certs
- module.identity_certs
- module.bootkube
- module.tectonic
- module.flannel_vxlan
- module.calico
- module.canal

Initializing provider plugins...

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other```

**Comment by kalmog** [comment]
@carlosaya Also, seems that you're using an older version of terraform. Please make sure your TF binary is version v0.11.1 and try again.

**Pull request merged**